### PR TITLE
setup fuzzing

### PR DESCRIPTION
**Pull request by ksurent**
this commit adds basic support for https://github.com/dvyukov/go-fuzz

the only package that's currently being fuzzed is ./pkg/rec

several things worth mentioning:

I added go-fuzz to go.mod  because go-fuzz-build puts it there every time it's invoked (i.e.
it adds uncommitted changes to go.mod)

to add a tool dependency, I followed the technique described in
https://github.com/go-modules-by-example/index/blob/master/010_tools/README.md

I added a separate make target for fuzzing, but I also made it a part of `make
check` with the intention of making it a part of the release process

### CODE REVIEW

**Comment by ksurent** [comment]
Fixes #24.

**Comment by ksurent** [comment]
Note that go-fuzz is currently broken in some cases with Go 1.15. See https://github.com/dvyukov/go-fuzz/issues/294 and https://github.com/golang/go/issues/40546.

This may come up if you're planning to update to 1.15 soon.

**Review by avereha — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -24,8 +24,13 @@ fmt:
 	gofmt -d -s .
 
 .PHONY: check
-check: all test lint
+check: all test lint fuzz
```
Comment (grzkv) [inline Makefile:27]: This is a frequently run target. Won't it be a bit too long with fuzzing?
I suggest we keep fuzzing as a separate target.

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,8 @@ cover.out
 test/sender/sender
 test/receiver/receiver
 
+# Fuzzing artefacts.
+corpus
+crashers
+suppressions
+*.zip
```
Comment (grzkv) [inline .gitignore:12]: IMO, this is a bit too generic of an ignore.

**Review by grzkv — CHANGES_REQUESTED** [review]
@ksurent Thanks for the PR! I have a couple of minor comments, but nothing major.

**Comment by grzkv** [comment]
> Note that go-fuzz is currently broken in some cases with Go 1.15. See [dvyukov/go-fuzz#294](https://github.com/dvyukov/go-fuzz/issues/294) and [golang/go#40546](https://github.com/golang/go/issues/40546).
> 
> This may come up if you're planning to update to 1.15 soon.

@ksurent Yes, we plan to update around now. Thanks for pointing this out!

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (ksurent) [reply .gitignore]: Done.

File: Makefile [hunk 3]
```diff
@@ -24,8 +24,13 @@ fmt:
...
 
 .PHONY: clean
 clean:
 	rm -f nanotube
+
+.PHONY: fuzz
+fuzz:
+	go-fuzz-build ./pkg/rec
+	go-fuzz -bin=rec-fuzz.zip
```
[the remark this answers is not in the document]
Comment (ksurent) [reply Makefile]: Done.

File: Makefile (hunk 1 above)
Comment (ksurent) [reply Makefile]: In the absence of a dedicated script for making a release, this looked like the second best place. But if you're saying that this is a target that's used in your development cycle, I agree it's too much.

Anyway, removed.

File: .gitignore (hunk 2 above)
Comment (ksurent) [reply .gitignore]: I've put everything fuzzing–related into a separate directory under test/. Won't clash with any zip files you may to commit to the repo in the future 😀

**Review by grzkv — APPROVED** [review]
@ksurent Awesome! Thanks a lot for the contribution.

File: pkg/rec/rec_fuzz.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package rec
```
Comment (dgryski) [inline pkg/rec/rec_fuzz.go:1]: `// +build gofuzz`

Then this function isn't part of the regular public API.

**Pull request merged**